src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs

```diff
@@ -1074,10 +1093,81 @@ private bool BelongsToGroup()
             return WorkspaceViewModel.Model.Annotations.ContainsModel(this.annotationModel);
         }
 
+        /// <summary>
+        /// Determines whether this group displays warning or error icons in its header.
+        /// </summary>
+        private void UpdateErrorAndWarningIconVisibility()
+        {
+            // No icons are displayed when the group is expanded / not collapsed.
+            if (IsExpanded)
+            {
+                GroupState = ElementState.Active;
+                return;
+            }
+
+            // Fetching all nodes inside the group.
+            List<NodeModel> nodeModels = Nodes
+                .OfType<NodeModel>()
+                .ToList();
+
+            // Subscribing to changes in the nodes' state. If a node's state 
+            // changes (e.g. to warning) we'll want to reflect this.
+            foreach (NodeModel nodeModel in nodeModels)
+            {
+                nodeModel.PropertyChanged += NodeModelOnPropertyChanged;
+            }
+
+            // Adding to this nodes from any nested groups.
+            nodeModels.AddRange
```

Comment:
so instead of checking all nested nodes couldn't you just check the `groupState` of the nested group?

Reply:
Would for sure be nicer - GroupState lives on the `AnnotationViewModel` rather than the `AnnotationModel` we have here.

Reply:
Adjusted this method to check the GroupState of the nested group now.